docs(self-managed): helm upgrade for orchestration cluster consolidation (#6543)
camunda/camunda-docs: docs/self-managed/installation-methods/helm/upgrade/helm-870-880.md

- Application metrics have changed — consult the release notes.

:::note
During the upgrade, do not change your architecture, configuration, or authentication method. This is not supported.
:::

## Perform the upgrade

### 1. Configure your 8.8 Helm chart

> Update your `values.yaml` with all required configuration changes.

Reviewer: ```suggestion
You **must update** your `values.yaml` to align with the 8.8 changes.
```

#### 1.1 Camunda Helm chart values file changes

Adjust your `values.yaml` based on the 8.8 changes.

- Multi-tenancy must be configured in **two** locations.
- TLS/self-signed certificates apply **only** in the Orchestration Cluster.
- Double-check any deprecated or renamed config values.

| **Old key**     | **New key**            | **Notes** |
| --------------- | ---------------------- | --------- |
| **Web Modeler** |                        |           |
| `postgresql`    | `webModelerPostgresql` |           |

> The consolidation of `Zeebe`, `Zeebe Gateway`, `Operate`, and `Tasklist` to `Orchestration Cluster` has three groups:

Reviewer: ```suggestion
The consolidation of `Zeebe`, `Zeebe Gateway`, `Operate`, and `Tasklist` into the new `Orchestration Cluster` requires three type of changes:
```

> **Group 1: Renamed keys**

Reviewer: ```suggestion
**Renamed keys - update references directly**
```

> Rename the top-level key `zeebe` to `orchestration`, and keys are used as they are.

Reviewer: ```suggestion
Rename the top-level key `zeebe` to `orchestration`. Sub-keys remain the same.
```

| **Old key** | **New key**     | **Notes** |
| ----------- | --------------- | --------- |
| **Zeebe**   |                 |           |
| `zeebe`     | `orchestration` |           |

> **Group 2: Moved keys**

Reviewer: ```suggestion
**Relocated keys - move under orchestration**
```

> The following keys should be moved from the old components to `orchestration`.

Reviewer: ```suggestion
The following keys must be moved from their old to the new `orchestration` section. Some sub-key names change slightly.
```

| **Zeebe Gateway**                   |                                      |                       |
| `zeebeGateway.ingress`              | `orchestration.ingress`              |                       |
| `zeebeGateway.contextPath`          | `orchestration.contextPath`          |                       |
| `zeebeGateway.service.restPort`     | `orchestration.service.httpPort`     | ⚠️ Different key name |
| `zeebeGateway.service.grpcPort`     | `orchestration.service.grpcPort`     |                       |
| `zeebeGateway.service.commandPort`  | `orchestration.service.commandPort`  |                       |
| `zeebeGateway.service.internalPort` | `orchestration.service.internalPort` |                       |
| **Operate**                         |                                      |                       |
| `operate.enabled`                   | `orchestration.profiles.operate`     |                       |
| **Tasklist**                        |                                      |                       |
| `tasklist.enabled`                  | `orchestration.profiles.tasklist`    |                       |

> **Group 3: Combined keys**

Reviewer: ```suggestion
**Merged keys - manually consolidate**
```

> The following keys, if used, should be manually combined and they require more attention as they are free-style inputs.

Reviewer: ```suggestion
These keys existed separately across components. They must now be combined under `orchestration`. Pay close attention, as most are free-form inputs.
```

| `zeebeGateway.sidecars`           |                                    |           |
| `operate.sidecars`                |                                    |           |
| `tasklist.sidecars`               |                                    |           |
| **ExtraVolumes**                  |                                    |           |
| `zeebe.extraVolumes`              | `orchestration.extraVolumes`       |           |
| `zeebeGateway.extraVolumes`       |                                    |           |
| `operate.extraVolumes`            |                                    |           |
| `tasklist.extraVolumes`           |                                    |           |
| **ExtraVolumeMounts**             |                                    |           |
| `zeebe.extraVolumeMounts`         | `orchestration.extraVolumeMounts`  |           |
| `zeebeGateway.extraVolumeMounts`  |                                    |           |
| `operate.extraVolumeMounts`       |                                    |           |
| `tasklist.extraVolumeMounts`      |                                    |           |

:::tip

> It's recommended to migrate to the new values structure using the new key `orchestration`, but it's possible to enable a compatibility layer in the Camunda Helm chart. That means the old structure from Camunda 8.7 (Helm chart 12.x.x) will work with Camunda 8.8 (Helm chart 13.x.x).

Reviewer: ```suggestion
It's recommended to migrate to the new values structure using the new key `orchestration` directly.

If you need more time, enable the compatibility layer so the old 8.7 values (Helm chart 12.x.x) still work in 8.8 (Helm chart 13.x.x).
```

> 

Reviewer: ```yaml
global:
  compatibility:
    core:
      enabled: true
```

> To enable the Camunda Helm chart compatibility layer, set `global.compatibility.core.enabled: true` in the Helm chart values file.

Reviewer: ```suggestion
```
